presto-feature-toggle/src/main/java/com/facebook/presto/features/config/FeatureConfiguration.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.features.config;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.presto.features.strategy.FeatureToggleStrategyConfig;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.facebook.airlift.json.JsonCodec.jsonCodec;
+
+public class FeatureConfiguration
+{
+    private static final JsonCodec<FeatureConfiguration> FEATURE_CONFIGURATION_JSON_CODEC = jsonCodec(FeatureConfiguration.class);
+
+    private String featureId;
+    private boolean enabled;
+    private String featureClass;
+    private boolean hotReloadable;
+    private List<String> featureInstances;
+    private String currentInstance;
+    private String defaultInstance;
+    private FeatureToggleStrategyConfig featureToggleStrategyConfig;
+
+    @JsonCreator
+    public FeatureConfiguration(
+            @JsonProperty String featureId,
+            @JsonProperty boolean enabled,
+            @JsonProperty boolean hotReloadable,
+            @JsonProperty String featureClass,
+            @JsonProperty List<String> featureInstances,
+            @JsonProperty String currentInstance,
+            @JsonProperty String defaultInstance,
+            @JsonProperty FeatureToggleStrategyConfig featureToggleStrategyConfig)
+    {
+        this.featureId = featureId;
+        this.featureClass = featureClass;
+        this.enabled = enabled;
+        this.hotReloadable = hotReloadable;
+        this.featureInstances = featureInstances;
+        this.currentInstance = currentInstance;
+        this.defaultInstance = defaultInstance;
+        this.featureToggleStrategyConfig = featureToggleStrategyConfig;
+    }
+
+    public static FeatureConfiguration fromJson(String json)
```

Comment:
Do we use this anywhere?

Reply:
No, but it's just an example of loading configuration from JSON. The first implementation of the feature toggle configuration was in JSON. I can remove this.

Reply:
removed